pylint/checkers/refactoring.py

```diff
@@ -261,12 +265,30 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
+        for part in node.body:
+            if isinstance(node.iter, astroid.node_classes.Name):
+                self._check_mutable_sequence_modified_in_loop(part,
+                                                              node.iter.name)
+
+    def _check_mutable_sequence_modified_in_loop(self, node, target_name):
+        if hasattr(node, 'body'):
+            for part in node.body:
+                self._check_mutable_sequence_modified_in_loop(part,
+                                                              target_name)
+        if (isinstance(node, astroid.node_classes.Expr) and
```

Comment:
Please add a testcase - create a custom iterable class with methods remove, delete and pop, check if error is not emitted for them.

This warning should be emitted only for built-in classes - in this case only lists.